js/src/viewer/manifestsPanel.js

```diff
@@ -48,13 +118,46 @@
           var _this = this;
 
           // handle subscribed events
+          // When the manifest selection panel is brought up or hidden
           _this.eventEmitter.subscribe('manifestsPanelVisible.set', function(_, stateValue) {
             _this.onPanelVisible(_, stateValue);
           });
 
+          // When a manifest is received
           _this.eventEmitter.subscribe('manifestReceived', function(event, newManifest) {
             _this.onManifestReceived(event, newManifest);
           });
+          
+          // When a collection is received
+          _this.eventEmitter.subscribe('collectionReceived', function(event, newCollection, parentUri, parentNodeId) {
+            _this.onCollectionReceived(event, newCollection, parentUri, parentNodeId);
+          });
+          
+          // When a collection failed to load
+          _this.eventEmitter.subscribe('collectionNotReceived', function(event, parentUri, parentNodeId) {
+            _this.onCollectionNotReceived(event, parentUri, parentNodeId);
+          });
+          
+          // When Mirador gets an explicit request to load a manifest from a URL
+          _this.eventEmitter.subscribe('ADD_MANIFEST_FROM_URL', function(event, url) {
+            // Add it if it has not been loaded before
+            if (_this.userManifests.indexOf(url) == -1) {
+              _this.userManifests.push(url);
+            }
+            // Shift to the "User-Loaded Manifests" folder
+            _this.treeElement.jstree('deselect_all');
+            _this.treeElement.jstree('select_node', 'user');
+          });
+          
+          // When Mirador gets an explicit request to load a collection from a URL
+          _this.eventEmitter.subscribe('ADD_COLLECTION_FROM_URL', function(event, url, source) {
+            _this.addCollectionFromUrl(url, null, true);
+          });
+          
+          // When Mirador gets a request to load an object (manifest or collection) from a URL (e.g. the "Add object from URL" form)
+          _this.eventEmitter.subscribe('ADD_OBJECT_FROM_URL', function(event, url, source) {
+            _this.addObjectFromUrl(url, source, true);
```

Comment:
There are too many hooks into ADD_MANIFEST_FROM_URL for me to safely rework them at this point. This is a compromise until we can re-architect Mirador 3.